### PR TITLE
(TST) Replacing Timestamp.now () in tests.

### LINKED ISSUE #44341: CI: Add CI check to exclude datetime/Timestamp.now in tests to prevent time dependent behaviors (e.g. DST change)
### Most CI test runs failed at 2021-11-07T02:05:19. when the clocks went back in the US

70 tests failed with `pytz.AmbiguousTimeError` at `Timestamp.now().tz_localize(dti.tz)` and `pd.Series([fill_value], dtype=fill_dtype)`:

[ambiguous_time_error_log.txt](https://github.com/pandas-dev/pandas/files/7492034/ambiguous_time_error_log.txt)

### PR DESCRIPTION
**Pull request by Mikhaylov-yv**
- [x] xref #44341
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Replacing Timestamp.now() in tests.
Left Timestamp.now () in the following tests:

```
scalar/timestamp/test_timestamp.py:290:        compare(Timestamp.now(), datetime.now())
scalar/timestamp/test_timestamp.py:291:        compare(Timestamp.now("UTC"), datetime.now(timezone("UTC")))
scalar/timestamp/test_timestamp.py:320:        compare(Timestamp.now(), datetime.now())
scalar/timestamp/test_timestamp.py:321:        compare(Timestamp.now("UTC"), datetime.now(tzutc()))
scalar/timestamp/test_constructors.py:480:        ts_from_method = Timestamp.now()
scalar/timestamp/test_constructors.py:484:        ts_from_method_tz = Timestamp.now(tz="US/Eastern")
```

### CODE REVIEW

File: pandas/tests/arrays/test_datetimelike.py [hunk 1]
```diff
@@ -76,6 +76,14 @@ def timedelta_index():
     return TimedeltaIndex(["1 Day", "3 Hours", "NaT"])
 
 
+@pytest.fixture
+def fixed_now_ts():
```
Comment (mroeschke) [inline pandas/tests/arrays/test_datetimelike.py:80]: This should just be defined once in `pandas/tests/conftest.py` so it doesn't have to be redefined in multiple places.

Also please add non-zero minute, second and microsecond

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: pandas/tests/arrays/test_datetimelike.py (hunk 1 above)
Comment (Mikhaylov-yv) [reply pandas/tests/arrays/test_datetimelike.py]: Yes, of course, thanks for the advice.

Used fixed_now_ts where it does not interfere with reading the code.

File: pandas/conftest.py [hunk 2]
```diff
@@ -1268,6 +1268,16 @@ def timedelta64_dtype(request):
     return request.param
 
 
+@pytest.fixture
+def fixed_now_ts():
+    """
+    Fixture emits fixed Timestamp.now()
+    """
+    return Timestamp(
+        year=2021, month=1, day=1, hour=12, minute=0, second=0, microsecond=0
```
Comment (mroeschke) [inline pandas/conftest.py:1277]: Per my previous suggestion, please make these _not_ zero. e.g. 4, 13, 22

**Comment by mroeschke** [comment]
Does this cover all the non-intentional cases of `now()`?

File: pandas/conftest.py (hunk 2 above)
Comment (Mikhaylov-yv) [reply pandas/conftest.py]: I fixed it

**Comment by Mikhaylov-yv** [comment]
@mroeschke 

> Does this cover all the non-intentional cases of `now()`?

Yes it covers all cases.

Where I was in doubt left `now()`

**Review by mroeschke — APPROVED** [review]
Thanks LGTM. If you're interested in a follow up, you could add a CI check to ensure `now` is only used in a certain file where `now` should be explicitly tested.

**Comment by Mikhaylov-yv** [comment]
@mroeschke Oh sure. But I don’t understand how to do it.

**Comment by mroeschke** [comment]
No problem can you merge master one more time @Mikhaylov-yv ?

**Comment by Mikhaylov-yv** [comment]
@mroeschke yes of course.

**Comment by mroeschke** [comment]
Thanks @Mikhaylov-yv. CI failures unrelated

**Pull request merged**